lib/fog/storage/openstack/requests/public_url.rb

```diff
@@ -21,6 +21,27 @@ def url
           "#{@scheme}://#{@host}:#{@port}#{@path}"
         end
       end
+
+      class Mock
+        # Get public_url for an object
+        #
+        # ==== Parameters
+        # * container<~String> - Name of container to look in
+        # * object<~String> - Name of object to look for
+        #
+        def public_url(container = nil, object = nil)
```

Comment:
Could you try enabling some of the tests we currently skip in `test/requests` to use this with `FOG_MOCK`?